urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/crontrolnode/rpc/TopologyServer.java

```diff
@@ -0,0 +1,49 @@
+package com.sinopec.crontrolnode.rpc;
+
+import java.net.MalformedURLException;
+import java.rmi.Naming;
+import java.rmi.RemoteException;
+import java.rmi.registry.LocateRegistry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.sinopec.cache.NodeInfoCache;
+import com.sinopec.dao.GroupInfoDao;
+import com.sinopec.impl.GroupInfoDaoImpl;
+
+public class TopologyServer implements Runnable {
+
+	private static final Log LOG = LogFactory.getLog(TopologyServer.class);
+
+	private int port;
+
+	private NodeInfoCache nodeInfoCache;
+
+	public TopologyServer(int port, NodeInfoCache nodeInfoCache) {
+		super();
+		this.port = port;
+		this.nodeInfoCache = nodeInfoCache;
+	}
+
+	public void startListen() {
+		try {
+			LocateRegistry.createRegistry(port);
+			GroupInfoDao groupInfoDao = new GroupInfoDaoImpl(nodeInfoCache);
+			Naming.rebind("rmi://127.0.0.1:" + port + "/topology", groupInfoDao);
+			LOG.info("CrontrolNode is lintening on port " + port + "[Waiting for Group connection]: starting");
+		} catch (RemoteException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		} catch (MalformedURLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Stack trace information reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. 

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).